### PR TITLE
Implemented encode.go

### PR DESCRIPTION
**Pull request by VeerChaurasia**

### CODE REVIEW

File: athena_abi/encode.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+	"math/big"
+)
+
+func GetEnumIndex(enumType StarknetEnum, enumKey string) (int, StarknetType, error) {
+	for idx, variant := range enumType.Variants {
+		if variant.Name == enumKey {
+			return idx, variant.Type, nil
+		}
+	}
+	return 0, nil, &InvalidAbiError{Msg: fmt.Sprintf("enum key %s not found in enum %s", enumKey, enumType.Name)}
```
Comment (Sh0g0-1758) [inline athena_abi/encode.go]: this is not an invalid abi error. you can just return an error string.

File: athena_abi/encode.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+}
+
+func EncodeCoreType(encodeType StarknetCoreType, value interface{}) ([]*big.Int, error) {
+	switch encodeType {
+	case U8, U16, U32, U64, U128, U256:
+		var bigIntValue *big.Int
+		switch v := value.(type) {
+		case int64:
```
Comment (Sh0g0-1758) [inline athena_abi/encode.go]: I believe we can remove this case ?

File: athena_abi/encode.go [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+			if len(v) > 2 && v[:2] == "0x" {
+				var ok bool
+				intEncoded, ok = new(big.Int).SetString(v[2:], 16)
+				if !ok {
+					return nil, &TypeEncodeError{Msg: fmt.Sprintf("invalid hex string: %s", v)}
+				}
+			} else {
+				return nil, &TypeEncodeError{Msg: fmt.Sprintf("hex strings must be 0x prefixed: %s", v)}
+			}
+		case int64:
```
Comment (Sh0g0-1758) [inline athena_abi/encode.go]: again this case is not required.

File: athena_abi/encode.go [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+			enumValue, ok := encodeValue.(map[string]interface{})
+			if !ok {
+				return nil, &TypeEncodeError{Msg: fmt.Sprintf("%v cannot be encoded into a StarknetEnum", encodeValue)}
+			}
+			if len(enumValue) != 1 {
+				return nil, &TypeEncodeError{Msg: fmt.Sprintf("enum value %v must have exactly one key-value pair", enumValue)}
+			}
+			var enumKey string
+			var enumValueInner interface{}
+			for k, v := range enumValue {
```
Comment (Sh0g0-1758) [inline athena_abi/encode.go:145]: why take a range ? 
In python : `enum_key, enum_value = enum_items[0]`

**Review by Sh0g0-1758 — CHANGES_REQUESTED** [review]

File: athena_abi/encode.go (hunk 2 above)
Comment (VeerChaurasia) [reply athena_abi/encode.go:22]: the int64 one???

**Review by Sh0g0-1758 — COMMENTED** [review]
LGTM, thanks for the contribution.

**Review by Sh0g0-1758 — APPROVED** [review]

**Pull request merged**